README.md

```diff
@@ -13,9 +13,14 @@ The following demos provide examples of how to use the tools available with [Ope
 * [Video on creating Elyra Pipelines and Superset Dashboard](https://youtu.be/TFgsR7UlcHA)
 
 
-## [Demo 2 - Automated Inference Pipeline and Dashboarding](notebooks/demo2/README.md)
+## [Demo 2 - Automated Inference Pipeline](notebooks/demo2/README.md)
 
 * [Steps to Create a JupyterHub image from the repository](notebooks/demo2/README.md#jupyterhub-image-setup-aicoe-ci-thoth)
 * [Automated inference pipeline](https://github.com/os-climate/aicoe-osc-demo/blob/master/notebooks/demo2/inference.pipeline)
+
+
+## [Demo 3 - ELT and Dashboarding](notebooks/demo3/README.md)
```

Comment:
It seems like this header is trying to link to a readme inside "demo3" but I don't believe such a directory exists in the project. Is this a typo or older code perhaps?

Reply:
This is an incomplete reversion of a change that I made because elsewhere in the code there was a Demo3 (focused on loading data into Trino) that was documented in the notebook (https://github.com/os-climate/aicoe-osc-demo/blob/master/notebooks/demo2/create_results_table.ipynb) but not fully separated out.  I separated it out into a stand-alone demo3 directory, and then after talking with @Shreyanand, agreed to un-separate it out.

In the process of restoring the README.md file, it is clear there are a few more loose ends hanging around:

- [ ] Numerous references to CL1, which should be migrated to CL2 (Trino, SuperSet, and JupyterHub)
- [ ] Very old Trino patterns that are not consistent with osc-ingest-tools

Since you've asked me to restore the output cells of the notebooks, I think the notebooks should run against the correct cluster to produce their output.  (I did all my work on the correct cluster--it's just that the documentation doesn't state that correctly.)  OK if I add those changes to completely clean things up, or should I make a new PR?

Reply:
Yeah I agree, the README does have some loose ends that need to be tied. I don't have a very strong preference but I'm leaning a bit towards having a separate PR for docs update, and keeping this one focused on the env var usage update. WDYT?

Reply:
Sounds good, thank you! So #169 would address changes in the docs and `create_results_table.ipynb`, right? Could you please add the outputs of the inference notebooks back in this PR?

Reply:
Oops.  I took care of the readme, but need to re-run notebooks.  Just a minute...

Reply:
Things are not going great in terms of re-running the notebooks to get output cells in sync.  I'm failing to get a start on CL1

```
Server requested
2022-07-01T20:23:03.930371Z [Normal] Successfully assigned odh-jupyterhub/jupyterhub-nb-michaeltiemannosc to ip-10-0-193-88.ec2.internal
2022-07-01T20:23:06Z [Normal] AttachVolume.Attach succeeded for volume "pvc-2e638620-739d-4fa4-b0cb-30ab2c2309c8"
2022-07-01T20:23:26Z [Normal] Add eth0 [10.131.6.233/23] from openshift-sdn
2022-07-01T20:23:26Z [Normal] Container image "quay.io/os-climate/aicoe-osc-demo@sha256:3a31ef1196e3dacb9ca18fbb9f6820328955441ded9af87f7fad90727b4ad4f0" already present on machine
2022-07-01T20:23:27Z [Normal] Created container notebook
2022-07-01T20:23:27Z [Normal] Started container notebook
2022-07-01T20:23:29Z [Warning] Back-off restarting failed container
```

and CL2 notebooks are missing fundamental things I'd expect from using the custom AICoE-demo starting point (it cannot find all the libraries rooted at 'src' and when I use `pipenv install --system` it over-fills my local disk and cannot run the notebooks.

Is something broken on the cluster side today?

Reply:
I'm also seeing this running the pipeline in a CL2 environment that's referencing a CL1 Trino endpoint:

```
Error making request
Unexpected token < in JSON at position 0
Check the JupyterLab log for more details
```

Except I have no clue how to check the JupyterLab log.  I've opened a `Console Log` window, but that shows me nothing.

Reply:
My problems are likely related in part to https://github.com/os-climate/os_c_data_commons/issues/184, which seems to have taken down pods on both CL1 and CL2 during this time period.

Reply:
Whatever was killing the pods has been fixed, and I've managed to get each of the notebooks to run in turn as expected.  But there are problems running the whole pipeline (as reported above).  I'm going to check in my current changes, with the following notes:

* I'm using the samples_1 directory (s3://redhat-osc-physical-landing-647521352890/aicoe-osc-demo/pipeline_run/samples_1/) which has a single PDF that requires about 10 minutes to process.  If I used the default directory with 145 reports, it would take 1450 minutes, or about 24 hours, which is dumb for simple CI/CD validation.
* The concept of a single fixed Trino table is questionable, but I don't want to get too deep into solving too many problems just to get the notebook outputs restored.
* What I'm about to commit are about the minimal changes I can think of to sanely restore notebook outputs.

Reply:
> and CL2 notebooks are missing fundamental things I'd expect from using the custom AICoE-demo starting point (it cannot find all the libraries rooted at 'src'

Yeah @Shreyanand and I recently ran into this problem as well. We've created an updated image, and are testing it as I comment. If all goes well, we'll create a new release so that the image on cl2 is also updated and the import error gets fixed.

Reply:
Which of the many squashing techniques listed among these answers is most appropriate for this particular case?

https://stackoverflow.com/questions/25356810/git-how-to-squash-all-commits-on-branch

Reply:
Also, I don't understand what I need to do to fix the precommit problems.  I had quite a chase trying to fix this the last time around, and my memory is that ultimately I ended up back where I started in terms of newlines and files, but only after the merry chase did pre-commit accept.  I'm not up for doing that again.

Reply:
> Which of the many squashing techniques listed among these answers is most appropriate for this particular case?

I'd typically just do `git rebase -i HEAD~3`, change the "pick" to "s" or "squash" for every commit other than the topmost commit, and then `git push --force-with-lease`

Reply:
The squashing almost worked, but failed because when I switched branches to what I thought was the correct one (PR #167), I did not incant the proper upstream or origin spells, leading to this:

```
michael$ git push --force-with-lease
fatal: The upstream branch of your current branch does not match
the name of your current branch.  To push to the upstream branch
on the remote, use

    git push github-desktop-MichaelTiemannOSC HEAD:automation_fixups

To push to the branch of the same name on the remote, use

    git push github-desktop-MichaelTiemannOSC HEAD

To choose either option permanently, see push.default in 'git help config'.
```

Please tell me the best way to recover from this.

Reply:
On the topic of precommit, the command `precommit` is not in my current path (neither in my notebook environment nor my desktop environment).

Reply:
Heh yeah this can be somewhat of a tricky business :) But I believe in this case you wanna push to the `automation_fixups` branch, so imo the first option should work i.e. `git push github-desktop-MichaelTiemannOSC HEAD:automation_fixups`

Reply:
> On the topic of precommit, the command `precommit` is not in my current path (neither in my notebook environment nor my desktop environment).

I see, in that case I think you'd have to manually install it by `pip install pre-commit`. But regardless, since precommit has been behaving strangely for your case, I think we can just go ahead and merge this PR for now, and not worry about prettifying the code.

Re: squashing, I see that github UI provides has a "squash and merge" option for PRs, so it seems like manually squashing commits won't be required :) Thanks for putting in the effort to clean it anyway tho, appreciate it!